src/test/scala/com/fulcrumgenomics/vcf/AssessPhasingTest.scala

```diff
@@ -0,0 +1,787 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package com.fulcrumgenomics.vcf
+
+import java.nio.file.Files
+
+import com.fulcrumgenomics.FgBioDef.{PathPrefix, PathToIntervals, PathToVcf, unreachable}
+import com.fulcrumgenomics.testing.{UnitSpec, VariantContextSetBuilder}
+import com.fulcrumgenomics.util.{Metric, NumericCounter}
+import com.fulcrumgenomics.vcf.Cigar.IlluminaSwitchErrors
+import dagr.commons.io.PathUtil
+import htsjdk.samtools.SAMFileHeader
+import htsjdk.samtools.util.{Interval, IntervalList}
+import htsjdk.variant.variantcontext.writer.{Options, VariantContextWriterBuilder}
+import htsjdk.variant.variantcontext.{GenotypeBuilder, VariantContext, VariantContextBuilder}
+import htsjdk.variant.vcf.VCFHeader
+
+object AssessPhasingTest {
```

Comment:
I'm going to skip reviewing these tests for now because I'm exhausted after reviewing everything so far, and I'm not sure how much value it will add.  Let me know if you want me to take a look later.